### PR TITLE
Prove Qwen3.5 MoE GGUF explicit-float runtime route

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

- pin `Flexan/kshitijthakkar-qwen3.5-moe-0.87B-d0.8B-GGUF@a9b8adbec2cc87479c772dac1944f313b4036c26` / `qwen3.5-moe-0.87B-d0.8B.Q2_K.gguf` (626,599,552 bytes, SHA-256 `e8a84df1a50ce65cf80c2b55bba8c6e80f913679fdf9e9439f2c3b52ef3145d5`)
- promote only the explicitly dequantized dense-MoE CPU route, with structured source-fidelity, target-storage, compute-mode, runtime, tokenizer, graph, and package evidence
- prove full logits, KV/convolution/recurrent state prefill and cached decode, replay/rollback/reorder, deterministic 20-token generation, package/report round-trips, tokenizer identity, and ORT GenAI 0.15.2 compatibility
- keep the default mixed-quantized route fail-closed and document the reduced checkpoint's publisher quality limitation and routed-expert tolerance
- correct Qwen3.5 M-RoPE channel interleaving independently of flat RoPE rotation

## Evidence

- graph SHA-256: `8c1aa1075cee03ffd5ce5bbd283ee88b2466e6c1d645f0a41e542230951d6f09`
- runtime package SHA-256: `bd69cf72acffd393a706bb012c23990acfdaf3a43e25ad919ef1148afac42a15`
- deterministic tokens: `[198, 198, 198, 321, 198, 198, 198, 198, 198, 198, 198, 198, 198, 198, 198, 198, 198, 198, 198, 198]`

## Validation

- promoted real-artifact runtime test: passed
- focused GGUF evidence/config/docs/YAML closure: 1,246 passed
- broad serial non-integration suite: 8,563 passed, 62 skipped, 12 deselected
- lint, generated-doc check, targeted type check, and independent code review: passed

### CODE REVIEW

File: tests/gguf_small_model_runtime_integration_test.py [hunk 1]
```diff
@@ -408,9 +438,71 @@ def _empty_promoted_state(
...
+        state_dict[f"{prefix}.down_proj"] = torch.stack(down)
+    state_dict = {name: value.float() for name, value in state_dict.items()}
+
+    hf_config = AutoConfig.from_pretrained(repository, revision=revision)
+    hf_config.dtype = torch.float32
+    reference = AutoModelForCausalLM.from_config(hf_config)
+    missing, unexpected = reference.load_state_dict(state_dict, assign=True)
+    assert not missing
+    assert not unexpected
+    return reference.eval()
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/gguf_small_model_runtime_integration_test.py]: Fixed in 3cf8023e. The same-value reference now explicitly converts all floating parameters and buffers to `torch.float32` after loading the exact dequantized GGUF state.

File: tests/gguf_small_model_runtime_integration_test.py [hunk 2]
```diff
@@ -1054,43 +1149,50 @@ def test_promoted_gguf_full_runtime_evidence(
         assert _sha256(asset_path) == sha256
 
     output_dir = tmp_path / "runtime-package"
-    assert shutil.disk_usage(tmp_path).free >= 2 * evidence.size
+    assert shutil.disk_usage(tmp_path).free >= max(2 * evidence.size, case.required_free_bytes)
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/gguf_small_model_runtime_integration_test.py:1151]: Checked against the repository formatter: Ruff intentionally keeps this assertion on one line, and `lintrunner` passes with that canonical formatting, so I retained the formatter-produced form.
